tarantool/crud.py

```diff
@@ -0,0 +1,84 @@
+"""
+This module provides API for interaction with the `crud`_ module.
+
+.. _crud: https://github.com/tarantool/crud/
+"""
+
+from tarantool.error import DatabaseError, ER_NO_SUCH_PROC
+
+
+class CrudResponse(object):
+    """
+    Contains response fields from the `crud`_ module that correspond 
+    to the Lua implementation.
+
+    .. _crud: https://github.com/tarantool/crud/
+    """
+
+    def __init__(self, response):
+        """
+        Sets response fields as in Lua implementation.
+
+        :param response: The response object of the crud module call.
+        :type response: :class:`~tarantool.response.Response`
+        """
+
+        if isinstance(response, dict):
+            for response_field_name in response.keys():
+                if isinstance(response_field_name, bytes):
+                    setattr(self, response_field_name.decode(), response[response_field_name])
+                else:
+                    setattr(self, response_field_name, response[response_field_name])
+        else:
+            raise RuntimeError('Unable to decode response to object due to unknown type')
+
+
+class CrudResult(CrudResponse):
+    """
+    Contains result's fields from result variable 
+    of crud module operation.
+    """
+
+
+class CrudError(CrudResponse):
+    """
+    Contains error's fields from error variable 
+    of crud module operation.
+    """
+
+
+def wrapCrudUndefinedProcedureError(e: DatabaseError):
```

Comment:
First, let's create a new error instead of messing with existing one. I think chaining mechanism may be really useful here: https://docs.python.org/3/tutorial/errors.html#exception-chaining

Reply:
Done, also added a case with error handling with code 42 (`ER_ACCESS_DENIED`). As far as I understand, now the exception chaining works as it should